cmd/actuary/server/server.go

```diff
@@ -34,68 +37,93 @@ var (
 			var report = outputData{Mu: &sync.Mutex{}, Outputs: m}
 			var reqList []check.Request
 
-			mux.HandleFunc("/request", func(w http.ResponseWriter, r *http.Request) {
+			// Get list of all nodes in the swarm via Docker API call
+			// Used for comparison to see which nodes have yet to be processed
+			ctx := context.Background()
+			cli, err := client.NewEnvClient()
+			if err != nil {
+				log.Fatalf("Could not create new client: %s", err)
+			}
+			nodeList, err := cli.NodeList(ctx, types.NodeListOptions{})
+			if err != nil {
+				log.Fatalf("Could not get list of nodes: %s", err)
+			}
+
+			// Send official list of nodes from docker client to browser
+			mux.HandleFunc("/getNodeList", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
yes, because getNodeList gets the official list of nodes -- which is compared to results, which grows as the results from each node comes in. When the list from getNodeList = list from results, then we know all tests are done processing.